Sources/FoundationEssentials/Data/Data+Base64.swift

```diff
@@ -0,0 +1,374 @@
+     this will throw a `Base64Error`.
+
+     - parameter bytes:      The Base64 bytes
+     - parameter output:     An OutputBuffer to be filled with decoded bytes
+     - parameter options:    Options for handling invalid input
+     - returns:              Whether decoding was successful
+     */
+    static func base64DecodeBytes(
+        _ bytes: borrowing BufferView<UInt8>,
+        _ output: inout OutputBuffer<UInt8>,
+        options: Base64DecodingOptions = []
+    ) throws {
+
+        // This table maps byte values 0-127, input bytes >127 are always invalid.
+        // Map the ASCII characters "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/" -> 0...63
+        // Map '=' (ASCII 61) to 0x40.
+        // All other values map to 0x7f. This allows '=' and invalid bytes to be checked together by testing bit 6 (0x40).
+        let base64Decode: StaticString = """
+\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\
+\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\
+\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\u{3e}\u{7f}\u{7f}\u{7f}\u{3f}\
+\u{34}\u{35}\u{36}\u{37}\u{38}\u{39}\u{3a}\u{3b}\u{3c}\u{3d}\u{7f}\u{7f}\u{7f}\u{40}\u{7f}\u{7f}\
+\u{7f}\u{00}\u{01}\u{02}\u{03}\u{04}\u{05}\u{06}\u{07}\u{08}\u{09}\u{0a}\u{0b}\u{0c}\u{0d}\u{0e}\
+\u{0f}\u{10}\u{11}\u{12}\u{13}\u{14}\u{15}\u{16}\u{17}\u{18}\u{19}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}\
+\u{7f}\u{1a}\u{1b}\u{1c}\u{1d}\u{1e}\u{1f}\u{20}\u{21}\u{22}\u{23}\u{24}\u{25}\u{26}\u{27}\u{28}\
+\u{29}\u{2a}\u{2b}\u{2c}\u{2d}\u{2e}\u{2f}\u{30}\u{31}\u{32}\u{33}\u{7f}\u{7f}\u{7f}\u{7f}\u{7f}
+"""
+        assert(base64Decode.isASCII)
+        assert(base64Decode.utf8CodeUnitCount == 128)
+        assert(base64Decode.hasPointerRepresentation)
+
+        let ignoreUnknown = options.contains(.ignoreUnknownCharacters)
+
+        var currentByte: UInt8 = 0
+        var validCharacterCount = 0
+        var paddingCount = 0
+        var index = 0
+
+        for base64Char in bytes {
+            var value: UInt8 = 0
+
+            var invalid = false
+            if base64Char >= base64Decode.utf8CodeUnitCount {
+                invalid = true
+            } else {
+                value = base64Decode.utf8Start[Int(base64Char)]
+                if value & 0x40 == 0x40 {       // Input byte is either '=' or an invalid value.
+                    if value == 0x7f {
+                        invalid = true
+                    } else if value == 0x40 {   // '=' padding at end of input.
+                        paddingCount += 1
+                        continue
+                    }
+                }
+            }
+
+            if invalid {
+                if ignoreUnknown {
+                    continue
+                } else {
+                    throw Base64Error.cannotDecode
+                }
+            }
```

Comment:
Why do not make something like that ?
```suggestion
            if invalid, ignoreUnknown  {
                    continue
                } else {
                    throw Base64Error.cannotDecode
                }
```

Reply:
That would be a different control flow. We want to either `continue` or `throw` if `invalid` is true, but  the rest of the loop must execute when `invalid` is false.